### ISSUE TITLE
Download NTM reference database error

### ISSUE THREAD

**Issue by khoahoc0508** [iqbal-lab-org/clockwork#93]
I can't download NTM reference database named LAWX00000000. Please give new download link. Thank you!

Console log:
Checking availability of https://www.ebi.ac.uk/ena/browser/api/fasta/LAWX00000000
ERROR: Record does not exist or is not available for accession provided
Error running: enaDataGet -d tmp.185201 -f fasta LAWX00000000 at /clockwork/scripts/download_tb_reference_files.pl line 189.

**Comment by martinghunt** [comment]
Thanks for reporting this. That genome was recently replaced with CP083405. The download script is updated.

I've also just updated the human genome download, because that had moved (you would have got an error on that as well otherwise).

(commits fe62f000f4eb478549e3a36f786f5be0c9a2b168 and 4a5d610bc4ddf1522cfb932fb3b47b0a2d5222c8)

**Comment by khoahoc0508** [comment]
Thanks for quickly update. Actually i find problem with human genome download and already download it directly to the folder. On other hand, while waitting for link update i found another reference database localed on ENA ftp of LAWX00000000 (attached below). Is that the same as CP083405? Thank you very much and best regard!
http://ftp.ebi.ac.uk/pub/databases/ena/wgs/public/law/LAWX01.fasta.gz

**Comment by iqbal-lab** [comment]
Martin is away this week on holiday, apologies!

**Comment by khoahoc0508** [comment]
I can wait for the answer.

**Comment by martinghunt** [comment]
When you try looking at LAWX00000000 in the ENA browser, it sends you to the updated version LAWX01000000.1. But there look to be problems with this new version in the ENA - eg the sample ID is "not flink to download the contigs is ftp://ftp.ebi.ac.uk/pub/databases/ena/wgs/suppressed/law/LAWX01.dat.gz. Note the "suppressed" in the URL.

**Comment by martinghunt** [comment]
(sorry, didn't mean to close the issue! Caught the wrong key on my keyboard by accident)

tl;dr: just use CP083405. It's the new LAWX00000000.

Here's all the details ...

LAWX00000000 was superseded by an updated version from the same authors. If you search in Genbank, it says the record has been removed and gives you the link to that old version, and also to its replacement.

The accession of the superseded version is LAWX01000000/CP083405. Those IDs both work fine with Genkbank. In the ENA browser: CP083405 is not found; LAWX00000000 sends you to LAWX01000000;  LAWX01000000 is there, but with problematic metadata and links (main record is here at the time of writing https://www.ebi.ac.uk/ena/browser/view/LAWX01000000)

It looks like the link you found (http://ftp.ebi.ac.uk/pub/databases/ena/wgs/public/law/LAWX01.fasta.gz) is, based on the name of the fasta file, the genome for LAWX01000000. However, the download perl script uses ENA's downloader `enaDataGet` (from here: https://github.com/enasequence/enaBrowserTools):
```
enaDataGet -f fasta LAWX01000000
Checking availability of https://www.ebi.ac.uk/ena/browser/api/fasta/LAWX01000000
ERROR: Record does not exist or is not available for accession provided
```
...and that is failing on LAWX01000000. This ties in with the metadata looking messed up in the browser. Ideally, I prefer to use `enaDataGet` instead of direct ftp URLs, because if the URLs change then an up to date version of the script `enaDataGet` should still work.

The result of all this was that I decided to use the CP083405 ID instead, which the script downloads from GenBank with no problems using efetch.

**Comment by khoahoc0508** [comment]
Clearly answer. Thank you very much!

**Comment by martinghunt** [comment]
No problems! Will close this now, as everything should download ok.
